### ISSUE TITLE
search profiles or custom search

### ISSUE THREAD

**Issue by AJLinn** [ArctosDB/arctos#5341]
Issue Documentation is http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html

**Describe the bug**
@dustymc created a nice and simple custom search page for UAM:EH: https://arctos.database.museum/uam_eh/search.cfm
I think when we recently updated the ctculture attribute list, we may have broken the connection - there's no drop-down/pick list anymore.

**To Reproduce**
Go to https://arctos.database.museum/uam_eh/search.cfm and enter a culture in the Culture of Origin or Culture of Use field.

**Expected behavior**
I should get a result that contains only items with that culture in the culture of origin field - e.g., Yup'ik should return 1,387 entries - now it's returning all records.

**Screenshots**
Using new search page:
<img width="1425" alt="Screenshot 2022-12-05 at 9 03 10 AM" src="https://user-images.githubusercontent.com/17605945/205709540-b9c968c0-86ec-4016-9b0f-e3f1d9bf5d35.png">

Using custom UAM:EH search page:
<img width="967" alt="Screenshot 2022-12-05 at 9 02 33 AM" src="https://user-images.githubusercontent.com/17605945/205709599-60aecd03-336d-41b4-aee2-b06c1514db7e.png">

<img width="967" alt="Screenshot 2022-12-05 at 9 02 51 AM" src="https://user-images.githubusercontent.com/17605945/205709630-2256a398-3c99-412b-9a00-ac3dd91ad773.png">

I was going to ask if we'd still be able to use this abbreviated/custom search page after the new search page goes to default mode. If not, I'll need to make some updates to my department website.

Priority: medium (but will become high as soon as the new search page becomes default)

**Comment by dustymc** [comment]
I can't recreate, but I'll revisit this after next release which will include hopefully-helpful tools (and hopefully will be out tonight).

Your "custom" form will not survive this migration. https://github.com/ArctosDB/arctos/issues/2745#issuecomment-1322942094 is intended to be a replacement - ultimately I'll need

* list of search options (and optionally values)
* list of ordered results columns
* name

I think that results in the same thing - a customized "portal" form - but in a much more sustainable manner. (Please let me know ASAP if that's crazy.... Also that could potentially include styling or ??????????, let me know if some sort of expansion needs discussion.)



We may also want to add search options (eg your most important attributes as their own things rather than part of a more general table) and potentially cached columns (the current implementation has performance issues). Both of those things come with a cost (bit of development, bit of maintenance, potentially-significant storage) and probably should have a dedicated discussion, but if that will result in significant usability benefits (and I think it will) then it's probably worth paying.

**Comment by AJLinn** [comment]
Thanks Dusty, this looks like exactly what we need.

> list of search options (and optionally values)

Is it possible to add the **Any taxon, ID, common name** field and the attribute **description** that would include all the previous fields on the old form.

> list of ordered results columns

(if possible)
- GUID
- Media
- Identified as
- Specific locality
- creator
- collector

Thanks so much! This is super helpful for our users who are often folks with no database experience!

**Comment by krgomez** [comment]
Would it be possible to make one of these profiles for searching the UAM:Art collection as well? We would want something very similar to what you have created for the EH collection. In addition to what's there now and what Angie has suggested above, art collection users would want to search:

- subject 
- materials
- event type (so as to distinguish between specific localities where works were created versus localities depicted)

The only field we wouldn't also need is culture of use, as it's not something the art collection records.

Having a portal like this would be SO helpful for people new to Arctos! I love how the page shows all of the EH items in the results to start with, so that a user can get a better sense of what they are searching. I think it would feel less intimidating to a new user or someone that doesn't have a good grasp of the collection. I also love how in this new form you can fine tune your results in the same interface that you started in.

**Comment by dustymc** [comment]
>what we need

Awesome - that's the big-picture question, is it worth adding this sort of complexity or does that still lead somewhere your users don't want to be? I think I'm hearing yes, "portals" are useful.

> possible to add the

Big-picture: yes, if you can search by it I can add it to a "portal."

(Slightly smaller picture, some things are more expensive than others and we might need to do some back-end shuffling before we can put some specific thing in front of users. I'm not sure you much need to care, other than I might ask if something's *really* important, and/or it might take me a bit longer to add some things than others.)

>results columns....Media

That gets into the above parentheticals a bit (it's expensive and that cost needs paid up front if we're going to put it in front of a large audience), but it also brings in another problem: We currently let anyone turn on 5000 rows, we allow relatively huge (in filesize - IIIF would be cool @mkoo ) thumbnails, there are many ways to melt about any browser in there. We need to do SOMETHING to limit that - maybe allow public users to only see 10 rows at a time, or limit rowcount in some portals, or ?? 

>Would it be possible to make one of these profiles

Yes, the design is that these are low-maintenance - and we can make lots of them without introducing untenable sustainability issues. I probably need to add a tiny bit of complexity if we're going to have more than ~dozens but even that should be straightforward.

>love how the page shows all of the EH items

Cool, then maybe I *won't* "find a way to delay initial search"

>fine tune your results in the same interface that you started in

Thanks! Yes, natively interactive is awesome, everything should work like that!

The task force (https://github.com/ArctosDB/arctos/issues/5324) which met this AM might have something to add, I'm not sure how to summon them as a group so @campmlc @Jegelewicz @ebraker

**Comment by krgomez** [comment]
If it's helpful to know, here is a list of the fields that would be useful in an art collection search portal:

- agents (creator, collector, copyright holder)
- identification
- taxon (fine to have the _any taxon, ID, common name_ field, but could also be nice to have a box to search title and a box to search object type)
- specific locality
- catalog number
- accession
- event type
- materials
- subject
- culture of origin

And here are the results fields that would be most useful to see as a default:

- GUID
- media
- identified as
- creator
- specific locality
- verbatim date
- full taxon name
- culture of origin

It is extremely helpful to see the media thumbnails in the results, so I hope we can keep that as an option somehow.

**Comment by dustymc** [comment]
Please comment in https://github.com/ArctosDB/arctos/issues/5354 if I missed anything. I think I got everything except event type - it's just not very "flattenable" I have no idea how it might be pulled out - always up for clever ideas...

**Comment by dustymc** [comment]
Next release will have cost-based page size calculations ranging from a maximum of 25 records per page for public users with media turned on to 5000 for users with accounts and without media turned on.

**Comment by dustymc** [comment]
# PLEASE READ 


Please do not share these too widely until https://github.com/ArctosDB/arctos/issues/5354 is implemented and tested. These links should work (and I'm doing this now so I can make adjustments if necessary), but performance will be cruddy until the cache updates can be made (Wednesday, I hope).


# / PLEASE READ 

Also note that you should be able to create redirects for these (or about anything else) if you don't like the URL for whatever reason, and I'm happy to help with that.

@AJLinn 

https://arctos.database.museum/search.cfm?sp=uam_eh&guid_prefix=UAM:EH

@krgomez

https://arctos.database.museum/search.cfm?sp=uam_art&guid_prefix=UAM:Art

@mkoo this is the 'partial customization' I mentioned somewhere along the way. These are hard-coded IF statements that leverage core functionality, so there's very little overhead. (If we get dozens of them that'll change, but I think will stay well within 'sustainable'.) These definitely do not have the capabilities or resilience of something that could be built against the API, but they don't involve creating or maintaining that 'something' either. I think they're pretty neat.

**Comment by AJLinn** [comment]
Wow, this is awesome! I know our users will feel more comfortable with this interface and I can easily write a how-to for public users on this page. Once we get the a-ok to put on our website, I'll update and we can request feedback from public users. Thank you @dustymc  for taking the time to construct these when you have so much more on your plate!

**Comment by mkoo** [comment]
late to the game here but wanted to say I LOVE what I'm seeing for a cultural collection search. As hard-coded IF's then I wonder if we can create profiles for cultural, archival and biocollections as default choices a user can access (completely public) from the home page. That's the goal, right?

From what I see so far for the Art/Ethno-focused collections, an Archives one would just be a subset or the same since the archives is defined pretty loosely and includes art work, things of cultural significance, and digital/physical records.

Also, would the Art/Ethno search include media searches too? Archive searches would be useful to have at least a media flag if present. I dont see that included at the moment.

**Comment by krgomez** [comment]
Yes, thank you so much @dustymc! I think it will mean the difference between the public searching the art collection and not, which is huge. I don't see the subject field in the art profile search options. Can this be added?

**Comment by dustymc** [comment]
> if we can create profiles for

If you can build it on the search page, I can create a profile for it. So yes to whatever follows, but it might involve [rebuilding the cache](https://github.com/ArctosDB/arctos/issues/5354) and such. (And there are probably things that can't readily be cached and so can't be in default results so maybe "yes" is a bit optimistic, but close.)

>That's the goal, right?

I'm not sure. If having some sort of "whatever portal" helps users then sure, or if tailoring something to the individual "whatever" collections is better I can do that too.

(And this is just search, so "all the whatevers, regardless of collection" is entirely possible too.)

I just rebuilt so the URLs don't need explicit collections, it would be useful to know the direction - I'll un-do that if we're heading somewhere more generic, eg if I might want to use the uam_art profile with some other collection.


>media flag 


<img width="165" alt="Screenshot 2022-12-13 at 9 11 17 AM" src="https://user-images.githubusercontent.com/5720791/207399031-6522a183-44eb-4ebb-a4e9-afcaf2039259.png">

check, save

<img width="486" alt="Screenshot 2022-12-13 at 9 11 31 AM" src="https://user-images.githubusercontent.com/5720791/207399047-9a1f7128-bed9-4917-891a-1b8e0d476eae.png">

set values

<img width="227" alt="Screenshot 2022-12-13 at 9 11 38 AM" src="https://user-images.githubusercontent.com/5720791/207399144-2e00bf6e-3ea8-42a8-ab12-7f3e2298f068.png">

<img width="292" alt="Screenshot 2022-12-13 at 9 11 48 AM" src="https://user-images.githubusercontent.com/5720791/207399157-18c2b3f1-a76d-4e65-821c-3723a3eff2d4.png">

will get a URL that can be adjusted, reused, built on to create a profile, whatever.

>subject field

Thx, I'll add it.

**Comment by AJLinn** [comment]
One thing I noticed with the custom page is if you hit the clear form button, you go back to the full (minimized) search page and the custom-selected fields go away. It's okay if that can't be avoided, we just need to let our users know to not use that button if they want to keep this custom fields in front of them.
Thank you!

**Comment by dustymc** [comment]
Yea I'm not sure how to handle that so I've been ignoring it, at least until some other stuff gets worked out.

Some users are never going to want to leave that portal.

Others are going to want to dump that for their previous customizations.

Some may want to start their customizations from the portal.

The rest will do something I haven't anticipated at all.

I can probably accommodate any of those use cases, but I'm not sure how to approach them all at once....

**Comment by AJLinn** [comment]
No worries - as I said, we can just have a warning on our instructions page so folks know to just manually remove previous search options rather than use the clear form button.

This gives our users options and a low-stress entry to the system. The more they use the interface the more comfortable they'll be with creating a profile and customizing their experience.

**Comment by dustymc** [comment]
I'm going to close this - I think we're done and it's getting messy - but please feel free to reopen or start a new issue if something still needs work.

@krgomez https://arctos.database.museum/search.cfm?sp=uam_art should be functional and performing adequately.

@AJLinn ditto https://arctos.database.museum/search.cfm?sp=uam_eh

**Comment by krgomez** [comment]
Thanks @dustymc! I notice that with the update I can now search subject, but it is also appearing as a results column. Can subject be removed from the default results? Also, when I click on subject, the documentation box just says pending. Maybe it just needs some time or something though.

Would it be possible to add verbatim date as an additional default results column?

Also wanted to mention that when I try further customizing my results, the media column moves all the way to the right, which isn't a big deal at all, but it seems like it should stay to the left.

**Comment by dustymc** [comment]
>subject, but it is also appearing as a results column. 

Next release.

>documentation

Yea, it's still using the old docs because I've been directed to keep the old form around and I don't want to rebuild a bunch of code just to toss it out then the old form dies for whatever reason. I'm not sure what to do about that, but I'll update at least this term from the code table.

>verbatim date as an additional default results

Next release.

>further customizing my results


I think we're not clear on how this works for a couple reasons, and it would be good if this somehow is understood by everyone.

First, see https://github.com/ArctosDB/arctos/issues/5341#issuecomment-1350169457 - I can do lots of things, but I can't do lots of things all at once.

At the moment, there's no 'further' customization. Customizing leaves the portal for either your previous customization or for the defaults. There's no reason a portal has to share any search or results columns with anything else - they can be entirely mutually exclusive.

Media is by default where it is because that's where @ArctosDB/search-ui-task-group put it. I can make another spreadsheet if you want to show up to the next meeting and move things around.

Media is wherever it is for you because you put it there; results order is user-controlled.

<img width="967" alt="Screenshot 2022-12-19 at 7 57 12 AM" src="https://user-images.githubusercontent.com/5720791/208467202-76245e55-8c8f-45a2-b615-a11b72d4a9ea.png">
